tests/unit/pages/lab/xeniumAnalyser.spec.tsx

```diff
@@ -139,6 +144,21 @@ describe('Xenium analyser', () => {
           });
         });
       });
+      describe('Equipment', () => {
+        it('should display an error message on blur when no equipment is selected', () => {
+          waitFor(async () => {
+            await selectFocusBlur('equipmentId');
+            expect(screen.getByText('Equipment is a required field')).toBeVisible();
```

Comment:
Just to understand, is this a better way to use it or await method where you do an action in act, outside of waitFor?

Reply:
It depends on the use case. In this scenario, we are waiting for elements to become visible, which can be contingent on an asynchronous operation. We need to wait for the elements to appear before taking action and asserting the results. An alternative approach is to use findBy with a promise, like this:
 _screen.findByText('equipmentId').then((el) => {
            el.blur(); 
            expect(screen.getByText('Equipment is a required field')).toBeVisible();
          });_
But also, _el.blur();_  is not advisable in most cases.  To ensure our tests closely mimic user behavior, it's better to use userEvent to interact with the element, as it provides a more realistic simulation of user actions."